app/Community/Components/UserProfileMeta.php

```diff
@@ -447,7 +447,7 @@ private function calculateAveragePointsPerWeek(User $user, bool $doesUserPreferH
             return 0;
         }
 
-        $weeksSinceFirstUnlock = $startingDate->diffInWeeks(Carbon::now());
+        $weeksSinceFirstUnlock = (int) $startingDate->diffInWeeks(Carbon::now(), true);
```

Comment:
Why are you adding `true` to all of these calls? The [documentation](https://carbon.nesbot.com/docs/#api-difference) suggests that makes it an absolute value, which will cause the logic on line 453 to never be true.

> All can take 2 optional arguments: date to compare with (if missing, now is used instead), and an **absolute boolean option** (false by default), it returns negative value when the instance the method is called on is greater than the compared date (first argument or now). **If set to true, that makes the method return an absolute value no matter which date is greater than the other**.

Reply:
My perception is the migration guide implies that's what needed to maintain the same logical behavior in Carbon 3:

> In Carbon 3, using `(int) $after->diffInSeconds($before, true)` or `(int) abs($after->diffInSeconds($before))` allows to get explicitly an absolute and truncated value, so the same result as in v2.

Reply:
For an _absolute_ value, yes.
> In Carbon 3, using (int) $after->diffInSeconds($before, true) or (int) abs($after->diffInSeconds($before)) allows to get explicitly **an absolute** and truncated value, so the same result as in v2.

Reply:
My interpretation of this doc is in Carbon 2 they're implied to be absolute. Local testing seems to support this.

For example, in the `User` model:
```php
return Carbon::now()->diffInMonths($this->created_at) < 1;
```

With Carbon 2 for user 255178, this returns 34.
With Carbon 3, the exact same code returns -34.853889037349.

Reply:
Ah, I see now. So these migration conversions maintain parity, but the logic might be wrong in a few use cases.